crates/bevy_asset/src/lib.rs

```diff
@@ -1,3 +1,143 @@
+//! In the context of game development, an "asset" is a piece of content that is loaded from disk and displayed in the game.
+//! Typically, these are authored by artists and designers (in contrast to code),
+//! are relatively large in size, and include everything from textures and models to sounds and music to levels and scripts.
+//!
+//! This presents two main challenges:
+//! - Assets take up a lot of memory; simply storing a copy for each instance of an asset in the game would be prohibitively expensive.
+//! - Loading assets from disk is slow, and can cause long load times and delays.
+//!
+//! These problems play into each other, for if assets are expensive to store in memory,
+//! then larger game worlds will need to load them from disk as needed, ideally without a loading screen!
+//!
+//! As is common in Rust, non-blocking asset loading is done using `async`, with background tasks used to load assets while the game is running.
+//! Bevy coordinates these tasks using the [`AssetServer`], storing each loaded asset in a strongly-typed [`Assets<T>`] collection.
+//! [`Handle`]s serve as an id-based reference to entries in the [`Assets`] collection, allowing them to be cheaply shared between systems,
+//! and providing a way to initialize objects (generally entities) before the required assets are loaded.
+//! In short: [`Handle`]s are not the assets themselves, they just tell how to look them up!
```

Comment:
```suggestion
//! In short: [`Handle`]s are not the assets themselves, they just tell how to look them up!
```
```suggestion
//! In short: [`Handle`]s are not the assets themselves, but serve as a "bookmark" to easily retrieve or manipulate the asset.
```
If people still remember what a bookmark is, of course :slightly_smiling_face:

Reply:
Hmm, I think that terminology is a bit confusing :)